addons/hr_expense/controllers/webmanifest.py

```diff
@@ -0,0 +1,45 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tools import file_open
+from odoo.addons.web.controllers import webmanifest
+
+
+class WebManifest(webmanifest.WebManifest):
+
+    def _get_service_worker_content(self):
+        body = super()._get_service_worker_content()
+        with file_open('hr_expense/static/src/service_worker.js', 'r', filter_ext=('.js',)) as f:
+            content = f.read()
+            # find the begin of the body event listener fetch
+            search_body_str = 'self.addEventListener("fetch", (event) => {\n'
+            len_end_fetch = len('});\n')
+            index_fetch = body.find(search_body_str) + len(search_body_str)
+            # extract the fetch to append it inside the existing event fetch
+            begin_fetch = "// begin fetch"
+            end_fetch = "// end fetch"
+            index_begin = content.find(begin_fetch) + len(begin_fetch) + len(search_body_str)
+            index_end = content.find(end_fetch) - len_end_fetch
+            fetch_method_content = content[index_begin:index_end]
+            body = body[:index_fetch] + fetch_method_content + body[index_fetch:]
+            # append the end of the file
+            body += content[index_end + len_end_fetch + len(end_fetch):]
+        return body
+
+    def _get_webmanifest(self):
+        manifest = super()._get_webmanifest()
+        if not manifest.get('share_target'):
+            manifest['share_target'] = {
+                'action': '/odoo?share_target=trigger',
+                'method': 'POST',
+                'enctype': 'multipart/form-data',
+                'params': {
+                    'title': 'title',
+                    'text': 'text',
+                    'url': 'url',
```

Comment:
Shouldn't those values be filled?

Reply:
Not used in your case, I forgot to cleanup these part thank